### PR TITLE
Builder refactoring including WithBucket.

### PR DESCRIPTION
**Pull request by BigUstad**

### CODE REVIEW

File: Minio/DataModel/Args.cs [hunk 1]
```diff
@@ -36,6 +38,7 @@ public static Hashtable CloneHashTable(Hashtable h)
         }
         public Args()
         {
+            this.RequestMethod = Method.HEAD;
```
Comment (kannappanr) [inline Minio/DataModel/Args.cs:41]: is this required?

File: Minio/DataModel/BucketArgs.cs [hunk 2]
```diff
@@ -13,18 +13,23 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+using RestSharp;
```
Comment (kannappanr) [inline Minio/DataModel/BucketArgs.cs:16]: is this required?

File: Minio/DataModel/BucketOperationsArgs.cs [hunk 3]
```diff
@@ -19,47 +19,57 @@
...
+            return this;
+        }
+
         public override RestRequest BuildRequest(RestRequest request)
         {
             request.XmlSerializer = new RestSharp.Serializers.DotNetXmlSerializer();
             request.RequestFormat = DataFormat.Xml;
             // ``us-east-1`` is not a valid location constraint according to amazon, so we skip it.
-            if (this.Location != "us-east-1")
+            if ( this.Location != "us-east-1" )
```
Comment (kannappanr) [inline Minio/DataModel/BucketOperationsArgs.cs:63]:
```suggestion
            if (this.Location != "us-east-1")
```

File: Minio.Examples/Cases/MakeBucketWithLock.cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
+/*
+ * MinIO .NET Library for Amazon S3 Compatible Cloud Storage, (C) 2017-2020 MinIO, Inc.
```
Comment (kannappanr) [inline Minio.Examples/Cases/MakeBucketWithLock.cs:2]:
```suggestion
 * MinIO .NET Library for Amazon S3 Compatible Cloud Storage, (C) 2020 MinIO, Inc.
```

File: Minio/DataModel/Args.cs (hunk 1 above)
Comment (BigUstad) [reply Minio/DataModel/Args.cs:41]: If it is not assigned, the request is HEAD.

**Comment by poornas** [comment]
are you planning to refactor all the bucket operations in this PR? other than that, looks good to me. Using abstract class here is a nice improvement.

**Review by kannappanr — APPROVED** [review]
LGTM

**Review by poornas — APPROVED** [review]

**Pull request merged**